src/librustc/ty/relate.rs

```diff
@@ -479,14 +478,9 @@ pub fn super_relate_tys<'a, 'gcx, 'tcx, R>(relation: &mut R,
         (&ty::Array(a_t, sz_a), &ty::Array(b_t, sz_b)) =>
         {
             let t = relation.relate(&a_t, &b_t)?;
-            assert_eq!(sz_a.ty, tcx.types.usize);
-            assert_eq!(sz_b.ty, tcx.types.usize);
```

Comment:
Seems we lost some asserts, why that?

Reply:
Because there are no more types in the "Unevaluated" case. I could reintroduce types, but there are only two uses: array lengths (`usize` guaranteed, pretty pointless to carry that information around), or `mir::Constant` (for associated constants), which already has a type field!

Reply:
Ack.